File: Binary Search/Peak_index_in_Mountain_array_1.cpp
class Solution {
public:
    int findPeakElement(vector<int>& nums) {
        if (nums.size()==0)
		    return -1;
        int l = 0;
	    int h = nums.size() - 1;
        int mid;

	    while (l + 1 < h) 
        {
		    mid = l + (h - l) / 2;
		    if (nums[mid - 1] > nums[mid + 1]) 
			    h = mid;
		    else    
    			l = mid;
	    }

	    if (nums[l] > nums[h])
		    return l;

	return h;
    }
};

File: Binary Search/Peak_index_in_Mountain_array_1.py
class Solution:
    def findPeakElement(self, nums: List[int]) -> int:
        l = 0
        h = len(nums) - 1
        while (l + 1) < h: 
            mid = l + int((h - l) / 2)
            if nums[mid - 1] > nums[mid + 1]:
                h = mid
            else:    
                l = mid
        if nums[l] > nums[h]:
            return l
        else:
            return h
        